yarn-project/p2p/src/services/libp2p/libp2p_service.ts

```diff
@@ -300,12 +299,14 @@ export class LibP2PService<T extends P2PClientType> extends WithTracer implement
     // Create request response protocol handlers
     const txHandler = reqRespTxHandler(this.mempools);
     const goodbyeHandler = reqGoodbyeHandler(this.peerManager);
+    const blockHandler = reqRespBlockHandler(this.worldStateSynchronizer.getL2BlockSource());
```

Comment:
This class already has a `L2BlockSource` dependency. We don't need to proxy through the world state.

Reply:
Oh fantastic